lib/qr_code/placement.ex

```diff
@@ -84,7 +84,7 @@ defmodule QRCode.Placement do
     |> Result.and_then(&add_alignments(&1, version, @alignment))
     |> Result.and_then(&add_dark_module(&1, version, @dark_module))
     |> Result.map(&fill_matrix_by_data(&1, size, encoding_data))
-    |> Result.map(fn matrix -> %{qr | matrix: matrix} end)
+    |> Result.and_then(fn matrix -> %{qr | matrix: matrix} end)
```

Comment:
Why is here`and_then`? Supplied lambda function does not return `{:ok, ...}` or `{:error, ...}` value.